lib/state_machine/types.hpp

```diff
@@ -18,9 +20,10 @@ struct TransitionKey {
 struct TransitionHasher {
```

Comment:
We would rather do the thing where we add a hash function in `std::` than use a hasher like this.